Improved Zally UX

semonte opened the pull request:
Link warnings to proper PSI elements.

![image](https://user-images.githubusercontent.com/1151536/46534264-40194f80-c8b0-11e8-9326-9ecff9ad2723.png)


Fixes #192

semonte committed:
Improved Zally UX

Link warnings to proper PSI elements.

Fixes #192

maxim-tschumak commented on examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java:
```diff
@@ -51,7 +53,7 @@ private LintingResponse lint(final PsiFile file) {
     }
 
-    public boolean shouldLint(final PsiFile file) {
+    private boolean shouldLint(final PsiFile file) {
```
Maybe, sine the latest change, `canLint()` would be a better name?

maxim-tschumak commented on examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java:
```diff
@@ -62,36 +64,45 @@ private boolean hasZallyUrl() {
+        Optional<Optional<PsiElement>> psiElement = Optional.ofNullable(jsonPointer)
+                .map(pointer -> pointer
+                        .replace("/", ".")
+                        .replace("~1", "/"))
```
We should probably also include `~0` to make the mapping complete. ([related StackOverflow question/answer](https://stackoverflow.com/questions/31483170/purpose-of-tilde-in-json-pointer))

maxim-tschumak reviewed (approved):
:+1:

semonte committed:
Unescape JSON pointer correctly

We also need to unescape "~" character.

Renamed method for readability.

Part of #192

semonte commented:
I have addressed the comments at https://github.com/zalando/intellij-swagger/pull/193/commits/07462bc72f0f6b02d608711a2aacfa0b71f61a25

maxim-tschumak commented:
:+1:
